aptos-move/framework/aptos-framework/sources/transaction_context.move

```diff
@@ -193,11 +193,15 @@ module aptos_framework::transaction_context {
     /// session counter, and local counter into a 128-bit value.
     /// Format: `<reserved_byte (8 bits)> || timestamp_us (64 bits) || transaction_index (32 bits) || session_counter (8 bits) || local_counter (16 bits)`
     /// The function aborts if the local counter overflows (after 65535 calls in a single session).
+    /// When compiled for testing, this function bypasses feature checks and returns a simplified counter value.
     public fun monotonically_increasing_counter(): u128 {
-        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
-        assert!(features::is_monotonically_increasing_counter_enabled(), error::invalid_state(EMONOTONICALLY_INCREASING_COUNTER_NOT_ENABLED));
-        let timestamp_us = timestamp::now_microseconds();
-        monotonically_increasing_counter_internal(timestamp_us)
+        if (__COMPILE_FOR_TESTING__) {
+            monotonically_increasing_counter_internal(1000000)
+        } else {
+            assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
```

Comment:
Question: do we need this flag? Context has been enabled for a while now, this is always true, or I am missing something?